packages/vector_graphics_compiler/lib/src/svg/masking_optimizer.dart

```diff
@@ -0,0 +1,362 @@
+  switch (fill) {
+    case path_ops.FillType.evenOdd:
+      {
+        return PathFillType.evenOdd;
+      }
+
+    case path_ops.FillType.nonZero:
+      {
+        return PathFillType.nonZero;
+      }
+  }
+}
+
+/// Converts vector_graphics Path to path_ops Path.
+path_ops.Path toPathOpsPath(Path path) {
+  final path_ops.Path newPath = path_ops.Path(toPathOpsFillTyle(path.fillType));
+
+  for (PathCommand command in path.commands) {
+    switch (command.type) {
+      case PathCommandType.line:
+        {
+          final LineToCommand lineToCommand = command as LineToCommand;
+          newPath.lineTo(lineToCommand.x, lineToCommand.y);
+        }
+        break;
+      case PathCommandType.cubic:
+        {
+          final CubicToCommand cubicToCommand = command as CubicToCommand;
+          newPath.cubicTo(
+              cubicToCommand.x1,
+              cubicToCommand.y1,
+              cubicToCommand.x2,
+              cubicToCommand.y2,
+              cubicToCommand.x3,
+              cubicToCommand.y3);
+        }
+        break;
+      case PathCommandType.move:
+        {
+          final MoveToCommand moveToCommand = command as MoveToCommand;
+          newPath.moveTo(moveToCommand.x, moveToCommand.y);
+        }
+        break;
+      case PathCommandType.close:
+        {
+          newPath.close();
+        }
+        break;
+    }
+  }
+
+  return newPath;
+}
+
+/// Converts path_ops Path to VectorGraphicsPath.
+Path toVectorGraphicsPath(path_ops.Path path) {
+  final List<PathCommand> newCommands = <PathCommand>[];
+
+  int index = 0;
+  final Float32List points = path.points;
+  for (path_ops.PathVerb verb in path.verbs.toList()) {
+    switch (verb) {
+      case path_ops.PathVerb.moveTo:
+        {
+          newCommands.add(MoveToCommand(points[index++], points[index++]));
+        }
+        break;
+      case path_ops.PathVerb.lineTo:
+        {
+          newCommands.add(LineToCommand(points[index++], points[index++]));
+        }
+        break;
+      case path_ops.PathVerb.cubicTo:
+        {
+          newCommands.add(CubicToCommand(
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+            points[index++],
+          ));
+        }
+        break;
+      case path_ops.PathVerb.close:
+        newCommands.add(const CloseCommand());
+        break;
+    }
+  }
+
+  final Path newPath = Path(
+      commands: newCommands, fillType: toVectorGraphicsFillType(path.fillType));
+
+  return newPath;
+}
+
+/// Gets the single child recursively,
+/// returns null if there are 0 children or more than 1.
+ResolvedPathNode? getSingleChild(Node node) {
+  if (node is ResolvedPathNode) {
+    return node;
+  } else if (node is ParentNode) {
+    if (node.children.length == 1) {
+      return getSingleChild(node.children.first);
+    } else {
+      return null;
+    }
+  } else {
+    return null;
+  }
+}
```

Comment:
Here and elsewhere, you can simplify this code a bit and remove some nesting if you prefer early returns. Consider:

```dart
ResolvedPathNode? getSingleChild(Node node) {
  if (node is ResolvedPathNode) {
    return node;
  }
  if (node is ParentNode && node.children.length == 1) {
    return getSingleChild(node.children.first);
  }
  return null;
```

Reply:
fixed.